bokehjs/test/devtools/devtools.ts

```diff
@@ -317,40 +317,40 @@ async function run_tests(ctx: TestRunContext): Promise<boolean> {
       }
 
       const all_tests = [...iter(top_level)]
-      const test_suite = all_tests
 
       if (randomize) {
         const random = new Random(seed)
         console.log(`randomizing with seed ${seed}`)
-        shuffle(test_suite, random)
+        shuffle(all_tests, random)
       }
 
       if (keyword != null || grep != null) {
         if (keyword != null) {
           const keywords = keyword
-          for (const [suites, test] of test_suite) {
+          for (const [suites, test] of all_tests) {
             if (!keywords.some((keyword) => description(suites, test).includes(keyword))) {
-              test.skip = true
+              test.omit = true
             }
           }
         }
 
         if (grep != null) {
           const regexes = grep.map((re) => new RegExp(re))
-          for (const [suites, test] of test_suite) {
+          for (const [suites, test] of all_tests) {
             if (!regexes.some((regex) => description(suites, test).match(regex) != null)) {
-              test.skip = true
+              test.omit = true
             }
           }
         }
       }
 
-      if (test_suite.length == 0) {
-        fail("nothing to test")
-      }
+      const test_suite = all_tests.filter(([, test]) => test.omit !== true)
+
+      const num_all_tests = all_tests.length
+      const num_test_suite = test_suite.length
```

Comment:
```suggestion
      const selected_tests = all_tests.filter(([, test]) => test.omit !== true)

      const num_all_tests = all_tests.length
      const num_selected_tests = selected_tests.length
```

It is somewhat challenging to read this if-statement longer down: `num_test_suite == num_all_tests`.